### PR TITLE
Karras VE, DDIM and DDPM flax schedulers

### PR DESCRIPTION
**Pull request by kashif**
for issue #475

### CODE REVIEW

File: src/diffusers/schedulers/scheduling_ddim_flax.py [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+            model_output = (sample - alpha_prod_t ** (0.5) * pred_original_sample) / beta_prod_t ** (0.5)
+
+        # 6. compute "direction pointing to x_t" of formula (12) from https://arxiv.org/pdf/2010.02502.pdf
+        pred_sample_direction = (1 - alpha_prod_t_prev - std_dev_t**2) ** (0.5) * model_output
+
+        # 7. compute x_t without "random noise" of formula (12) from https://arxiv.org/pdf/2010.02502.pdf
+        prev_sample = alpha_prod_t_prev ** (0.5) * pred_original_sample + pred_sample_direction
+
+        if eta > 0:
+            key = random.PRNGKey(seed)
```
Comment (patrickvonplaten) [inline src/diffusers/schedulers/scheduling_ddim_flax.py:248]:
```suggestion
            key = jnp.split(key, num=1)
```
https://jax.readthedocs.io/en/latest/_autosummary/jax.random.split.html

File: src/diffusers/schedulers/scheduling_ddim_flax.py [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+
+    def step(
+        self,
+        state: DDIMSchedulerState,
+        model_output: jnp.ndarray,
+        timestep: int,
+        sample: jnp.ndarray,
+        eta: float = 0.0,
+        use_clipped_model_output: bool = False,
+        seed: int = 0,
```
Comment (patrickvonplaten) [inline src/diffusers/schedulers/scheduling_ddim_flax.py:177]: Let's use a PRNGKey here

**Review by patrickvonplaten — APPROVED** [review]
Cool looks good to me! What do you think @anton-l ?

**Review by anton-l — APPROVED** [review]
Looks clean!

**Pull request merged**